plugins/shapefile/main.py

```diff
@@ -116,11 +110,21 @@ def calculate_stats(self):
                         if 'unique_values' not in misc:
                             misc['unique_values'] = []
 
-                        misc['unique_values'].append(value)
+                        if value not in misc['unique_values']:
+                            misc['unique_values'].append(value)
 
         self._stats = all_stats
 
+        with open(self.path.replace('.shp', '.json'), 'w') as f:
+            save = {}
+            for var, data in all_stats:
+                save[var] = data.to_dict
+            json.dump(save, f)
+
+    def get_num_features(self):
+        return self._num_features
+
     def get_features(self, date=None):
         with fiona.open(self.path, 'r') as shp:
-                features = [f for f in shp]
-        return features
+            for f in shp:
```

Comment:
I like the first option better. I didn't know `yield from <object>` was a thing. Cool!